core-java/src/test/java/com/baeldung/system/WhenDetectingOS.java

```diff
@@ -0,0 +1,25 @@
+package com.baeldung.system;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.baeldung.system.DetectOS.OperatingSystem;
+
+public class WhenDetectingOS {
+
+    @Test
+    public void whenUsingSystemProperty_shouldReturnOS() {
```

Comment:
Why is this working on Travis if it's linux-based?
After we sort it out, the test should be ignored so that it doesn't affect the build on various envs

Reply:
@pivovarit I am a bit lost here. I went through the build logs and didn't find anything related to this test case. I am under the impression that this didn't run otherwise the build should have failed. Can you help me out a bit? I will make the changes as suggested and ignore the tests in the test class so that it does lead to any build issues. I hope that should be fine?

Reply:
I think you should try renaming the class to something that ends with "*Test" - this is how surefire picks up test classes